src/spatial/traversal/IterationDiffAnalyzer.scala

```diff
@@ -0,0 +1,141 @@
+package spatial.traversal
+
+import argon._
+import spatial.lang._
+import spatial.node._
+import spatial.util.spatialConfig
+import spatial.util.modeling._
+import spatial.metadata.control._
+import spatial.metadata.access._
+import spatial.metadata.memory._
+import spatial.metadata.bounds.Expect
+
+case class IterationDiffAnalyzer(IR: State) extends AccelTraversal {
+
+
+  private def visitInnerControl(lhs: Sym[_], rhs: Op[_]): Unit = {
+    dbgs(stm(lhs))
+    val blks = rhs.blocks.map{block => latencyAndInterval(block) }
+    val latency = blks.map(_._1).sum
+    val interval = (1.0 +: blks.map(_._2)).max
+    dbgs(s" - Latency:  $latency")
+    dbgs(s" - Interval: $interval")
+    lhs.bodyLatency = latency
+    lhs.II = lhs.userII.getOrElse(interval)
+  }
+
+  private def findCycles(lhs: Sym[_], ctrl: Control[_]): Unit = {
+    ctrl.bodies.foreach{body =>
+      body.blocks.foreach{case (iters, block) =>
+        val stms = block.stms
+        val cycles = findAccumCycles(stms.toSet).accums
+        if (cycles.nonEmpty) {
+          dbgs(s"\n\nFound cycles in $lhs ($iters): ")
+          cycles.foreach{c => dbgs(s"  $c")}
+          cycles.collect{case AccumTriple(mem,reader,writer) if (mem.isLocalMem && reader != writer) => 
+
+            val read = reader.affineMatrices.head.matrix
+            val write = writer.affineMatrices.head.matrix
+            val diff = write - read
+            if (iters.nonEmpty) {
+              val stride = iters.last.ctrStep match {case Expect(c) => Some(c.toInt); case _ => None}
+              val par = iters.last.ctrParOr1
+              if (stride.isDefined) {
+                // iterDiff between iters
+                val thisIterReads  = List.tabulate(par){i => read.incrementConst(i*stride.get)}  // TODO: Replace this with access.affineMatrices.map(_.matrix)?
```

Comment:
If I understand correctly, here I am trying to manually create the Seq of access matrices that will be unrolled by adding stride to the `c` column for each par, but _.affineMatrices is a Seq that already contains this?